docs/how_to_guide/handling_composite_bases.md

```diff
@@ -0,0 +1,156 @@
+add.basis1.basis2.n_basis_funcs
+```
+
+However, if you associated a label to the basis, you can use it to get the corresponding basis element.
+
+```{code-cell} ipython3
+add["input2"]
+```
+
+And its parameters can be easily accessed.
+
+```{code-cell} ipython3
+add["input2"].n_basis_funcs
+```
+
+This works for any sub-element, including the one that are composite.
+
+```{code-cell} ipython3
+# get input1 + input2
+add["(input1 + input2)"]
+```
+
+Note that the label of this composite basis is assigned automatically. You can overwrite that with a custom label.
+
+```{code-cell} ipython3
+add["(input1 + input2)"].label = "my_custom_label"
+add
+```
+
+A label can be specified at initialization if the composite basis is defined directly.
+
+```{code-cell} ipython3
+nmo.basis.AdditiveBasis(
+    nmo.basis.BSplineEval(5),
+    nmo.basis.MSplineEval(5), 
+    label="my_custom_label"
+)
+```
+
+And if you are asking yourself what happens when two bases with the same label are composed, well, this results in an error.
+This guarantees that the labels are always unique and you can always retrieve a basis using its label.
+
+```{code-cell} ipython3
+:tags: [raises-exception]
+
+nmo.basis.BSplineEval(5, label="x") + nmo.basis.MSplineEval(5, label="x")
+```
+
+## Get and Set Composite Basis Parameters
+
+When working with composite bases, often times one wants to re-configurate specific components. Again, the easiest way to achieve this is labeling each element and using the label to retrieve the basis.
+
+```{code-cell} ipython3
+# get the basis function parameter
+print(add["input2"].n_basis_funcs)
+
+# set a new value for the parameter
+add["input2"].n_basis_funcs = 8
+print(add["input2"].n_basis_funcs)
+```
+
+This change is reflected on the composite basis.
+
+```{code-cell} ipython3
+# check that the input2 basis has now 8 basis funcs
+add
+```
+
+### Cross-validating Composite Basis Parameters
+Another way to get and set the basis parameter is via the `get_params` and `set_params` methods.
+
+The `get_params` method returns a dictionary, containing all the parameters. The dictionary keys start with the basis label, followed by a double underscore and the name of the parameter.
+
+```{code-cell} ipython3
+add.get_params()
+```
+
+Each of the key can be used as keyword argument to the `set_method` which in turns sets one or more of the parameter values.
+
+```{code-cell} ipython3
+add.set_params(input3__order=3, input1__bounds=(-1,1))
+```
+
+:::{admonition} Grid definition
+:class: info
+
+The parameter keys retrieved by `get_params` are the one needed to define a parameter grid when cross-validating your hyper-parameters with scikit-learn. Learn how to cross-validate basis parameters using [pipelines](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html) with [this notebook](sklearn-how-to). 
+:::
+
+
+Note that if you don't provide a label, basis class name is used to construct the keys.
```

Comment:
From here to the end belongs in the second section, I think. it's not about sklearn